crates/forge/src/cmd/create.rs

```diff
@@ -33,10 +38,99 @@ use foundry_config::{
-
+use std::{
+    borrow::Borrow, collections::BTreeMap, marker::PhantomData, path::PathBuf, sync::Arc,
+    time::Duration,
+};
 merge_impl_figment_convert!(CreateArgs, build, eth);
 
+/// Finds a contract in the artifacts by its bytecode hash
+fn find_contract_by_hash(output: &ProjectCompileOutput, target_hash: &str) -> Option<Bytes> {
+    for (_contract_name, artifact) in output.artifacts() {
+        if let Some(bytecode) = artifact.get_bytecode_bytes() {
+            let bytecode_bytes = bytecode.into_owned();
+            if !bytecode_bytes.is_empty() {
+                // Calculate keccak256 hash of the bytecode
+                use alloy_primitives::keccak256;
+                let calculated_hash = hex::encode(keccak256(&bytecode_bytes));
+
+                // Normalize both hashes by removing 0x prefix for comparison
+                let normalized_target = target_hash.trim_start_matches("0x");
+                let normalized_calculated = calculated_hash.trim_start_matches("0x");
+
+                if normalized_calculated == normalized_target {
+                    return Some(bytecode_bytes);
+                }
+            }
+        }
+    }
+    None
+}
+
+/// Handles factory dependencies for a contract deployment
+async fn handle_factory_dependencies(
+    output: &ProjectCompileOutput,
+    config: &Config,
+    private_key: &str,
+) -> Result<()> {
+    // Collect all factory dependencies from all contracts
+    let mut all_dependencies = BTreeMap::new();
+
+    for (_id, contract) in output.artifact_ids() {
+        if let ArtifactExtras::Resolc(extras) = &contract.extensions {
+            if let Some(factory_dependencies) = &extras.factory_dependencies {
+                for (bytecode_hash, contract_name) in factory_dependencies {
+                    all_dependencies.insert(bytecode_hash, contract_name);
+                }
+            }
+        }
+    }
+
+    if all_dependencies.is_empty() {
+        return Ok(());
+    }
+
+    // Get RPC URL for factory dependency uploads
+    let rpc_url = config.get_rpc_url_or_localhost_http()?;
+
+    // Upload each factory dependency
+    for (hash, name) in all_dependencies {
+        // Try to find the contract by hash directly (skip name-based lookup)
+        let bytecode = find_contract_by_hash(output, hash);
+
+        if let Some(bytecode) = bytecode {
+            // Skip child contracts (those with 0x3c04 prefix)
+            let bytecode_slice = bytecode.as_ref();
+            if bytecode_slice.len() >= 2 && bytecode_slice[0] == 0x3c && bytecode_slice[1] == 0x04 {
+                continue;
+            }
+
+            // Upload factory dependency using upload_child_contract_alloy
+            let scaled_encoded_bytes = bytecode.encode();
+            let storage_deposit_limit = Compact(10000000000u128);
+            let encoded_storage_deposit_limit = storage_deposit_limit.encode();
+            let combined_hex = "0x3c04".to_string() +
```

Comment:
It should be based on node metadata. Pallet index may be different between runtimes